src/jdk.naming.dns/share/classes/com/sun/jndi/dns/DnsClient.java

```diff
@@ -436,13 +437,17 @@ private byte[] doUdpQuery(Packet pkt, InetAddress server,
                 // Packets may only be sent to or received from this server address
                 InetSocketAddress target = new InetSocketAddress(server, port);
                 udpChannel.connect(target);
-                int pktTimeout = (timeout * (1 << retry));
+                // use 1L below to ensure conversion to long and avoid potential
+                // integer overflow (timeout is an int).
+                // no point in supporting timeout > Integer.MAX_VALUE, clamp if needed
```

Comment:
if I have read this correctly,  timeout is of type int, thus int Math.clamp(int, int, int) is being called returning type int and promoting to long. Are the any side effects to consider here?  And as timeoutLeft (or remainingTimeout) and pktTimeout were both int and now is type long, then why have timeout declared as type  int ?

Reply:
If I'm not mistaken here  it's `Math.clamp(long, long, long)` which is called - because `timeout * (1L << retry)` is a long. We could make it more obvious by using `0L` instead of `0` too.

Reply:
thanks for the clarification ... yes indeed, I didn't see the 1L as the original was (timeout * (1 << retry))
BUT I should have read the comment more precisely(small screens and no glasses !!)